### ISSUE TITLE
Code editor 2: Make instance variables in global functions show up as unset color instead of global color

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#6906]
### Is your feature request related to a problem?

Since #6809 was closed as intentional, I'm submitting this FR instead to request it be, well, not intentional. Having these variables show up in global color could be quite confusing, since it makes it seem like the variables *are* globals. It also is a coloring style otherwise only possible through globalvar (which is deprecated), since globals are supposed to always have `global.` in front. 

### Describe the solution you'd like

Instead of having feather show that any variables set in a global function must be globals, instead have it go for a more middle ground "I don't know what you're intending", and have the variables show up in grey, the color for variables that are unset or unknown.

### Additional context

There was a discussion about this in Gamemaker Kitchen

**Comment by DragoniteSpam** [comment]
if the whole thing was being written from scratch, this might be okay. but GMS2 users have 8 years of magenta variables being "global variables" and suddenly seeing magenta being used for something completely different is needlessly confusing

**Comment by Totobal5** [comment]
I completely agree, seeing magenta color where there is no global variable is weird and I don't like it.

**Comment by zreedy** [comment]
This has been changed since this ticket was filed. Closing.

**Comment by Emc1923** [comment]
Closing as per comment.
